File: backend/chat/src/main/java/com/ssafy/s12p21d206/achu/chat/domain/Message.java
package com.ssafy.s12p21d206.achu.chat.domain;

import com.ssafy.s12p21d206.achu.chat.domain.user.ChatUser;
import java.time.LocalDateTime;

public record Message(
    Long id,
    Long chatRoomId,
    String content,
    ChatUser sender,
    MessageType type,
    LocalDateTime timestamp) {}

File: android/app/src/main/java/com/ssafy/achu/data/model/chat/Message.kt
package com.ssafy.achu.data.model.chat

import kotlinx.serialization.Serializable

@Serializable
data class Message(
    val content: String,
    val id: Int,
    val senderId: Int,
    val timestamp: String,
    val type: String,
    val isMine: Boolean = false
)